example/integrations/tracker-red-box-ltd.tsx

```diff
@@ -34,63 +25,55 @@ const Icon: React.FC = () => (
   </svg>
 )
 
-// ensure that the tracker script will be initialized once in runtime
-let wasInitialized = false
+const ScriptInjector: React.FC = () => {
+  useEffect(() => {
+    window.rbltd = window.rbltd || []
 
-const WrapperComponent: React.FC = ({ children }) => {
-  React.useEffect(() => {
-    if (!wasInitialized) {
-      createRedBoxTracker()
-      wasInitialized = true
+    return () => {
+      delete window.rbltd
     }
-  }, [])
+  })
+
+  useScript('/red-box-ltd.js', { id: 'red-box-ltd' })
 
-  return (
-    <div
-      style={{ border: '3px solid #C21515' }}
-      data-testid="consent-manager-wrapping-component"
-    >
-      {children}
-    </div>
-  )
+  return null
 }
 
-export function useRedBoxLtd(): Tracker {
+interface RedBoxLtdTracker extends Array<Array<String>>, Tracker {}
+
+// For usage non-react context when tracking page views (Docusaurus, Gatsby, ...)
+// @todo we need to save-guard this by checking if integration is actually enabled
+export function getRedBoxLtd() {
+  return window.rbltd
+}
+
+export function useRedBoxLtd(): RedBoxLtdTracker | null {
   const [isEnabled] = useDecision('red-box-ltd')
+  const [tracker, setTracker] = React.useState(null)
 
-  const redBoxLtdInterface = React.useMemo(() => {
-    if (!isEnabled) {
-      return {
-        trackEvent: () => {},
-        trackPageView: () => {},
-      }
+  React.useEffect(() => {
+    if (isEnabled && !tracker) {
+      locateTracker('rbltd', setTracker)
```

Comment:
relevant, this ensures react can track to the correct object as soon its available